### ISSUE TITLE
[Bug Report] Jacobian calculation for quadruped robot

### ISSUE THREAD

**Issue by catachiii** [NVIDIA-Omniverse/orbit#267]
### Describe the bug

The Jacobian from `ArticulationView` for quadruped is wrong.

### Steps to reproduce

I followed the documentation [here](https://isaac-orbit.github.io/orbit/source/setup/sample.html) after setting up the environment. I tried to run a minimal example to test the environment of quadruped by

`./orbit.sh -p source/standalone/environments/zero_agent.py --task Isaac-Velocity-Flat-Unitree-Go2-v0 --num_envs 32`

I put a breakpoint at `step` function in `rl_task_env.py` right after `if self.sim.has_gui() self.sim.render()` and try to access the Jacobian through

`jacobian = self.scene.articulations['robot'].root_physx_view.get_jacobians()`

I was trying to find the Jacobian mapping from ee_pos to joints, so I wanted to see what the value is by

`jacobian[0, 4, 0:3, -12:]`

which means the first env, the fifth body (FL_foot), position, 12 dofs.

But I got a matrix with zeros almost everywhere, as shown below.

```
tensor([[ 0.0000e+00,  0.0000e+00, -1.2096e-04,  0.0000e+00,  0.0000e+00,
          0.0000e+00,  0.0000e+00,  0.0000e+00,  0.0000e+00,  0.0000e+00,
          0.0000e+00,  0.0000e+00],
        [ 0.0000e+00,  0.0000e+00,  2.1096e-05,  0.0000e+00,  0.0000e+00,
          0.0000e+00,  0.0000e+00,  0.0000e+00,  0.0000e+00,  0.0000e+00,
          0.0000e+00,  0.0000e+00],
        [ 0.0000e+00,  0.0000e+00,  1.9390e-03,  0.0000e+00,  0.0000e+00,
          0.0000e+00,  0.0000e+00,  0.0000e+00,  0.0000e+00,  0.0000e+00,
          0.0000e+00,  0.0000e+00]], device='cuda:0')
```
which of course is wrong according to the kinematic chain of quadruped robots. 

I also tested with anymal-c, it has the same issue. 

### System Info

Describe the characteristic of your environment:

- Commit: main
- Isaac Sim Version: 2023.1.1
- OS: Ubuntu 22.04
- GPU: RTX 4090
- CUDA: 12.3
- GPU Driver: 545.23.08

### Additional context

No

### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have checked that the issue is not in running Isaac Sim itself and is related to the repo (not 100% sure)

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [ ] Outputting correct Jacobian for floating-based robots.

**Comment by hhansen-bdai** [comment]
Hello @catachiii, thanks for opening this issue! :smile:  However, I think it would be better if you changed this to the Bug Report template. If you fill out the template with the information it requests, we will address this as soon as we can :+1:

**Comment by catachiii** [comment]
Sure I can do that 😄

**Comment by Mayankm96** [comment]
@catachiii 

How did you obtain the link index for the Jacobians. We identified an issue when you try to work directly with PhysX articulation view. The indexing of body names inside the articulation view is not the same as that in the body view.

For body names in the ArticulationView, we follow the rigid prim view ordering:

```python
    @property
    def body_names(self) -> list[str]:
        """Ordered names of bodies in articulation (through rigid body view)."""
        prim_paths = self.body_physx_view.prim_paths[: self.num_bodies]
        return [path.split("/")[-1] for path in prim_paths]
```

However, in the articulation view, the body ordering is different:

```python
   @property
    def body_names(self) -> list[str]:
        """Ordered names of bodies in articulation (through articulation view)."""
        return self._root_physx_view.shared_metatype.link_names
```

This yields two different listing (for example, for ANYmal):

```
Link names through body view: ['base', 'LF_HIP', 'LF_THIGH', 'LF_SHANK', 'LF_FOOT', 'LH_HIP', 'LH_THIGH', 'LH_SHANK', 'LH_FOOT', 'RF_HIP', 'RF_THIGH', 'RF_SHANK', 'RF_FOOT', 'RH_HIP', 'RH_THIGH', 'RH_SHANK', 'RH_FOOT']
Link names through articulation view: ['base', 'LF_HIP', 'LH_HIP', 'RF_HIP', 'RH_HIP', 'LF_THIGH', 'LH_THIGH', 'RF_THIGH', 'RH_THIGH', 'LF_SHANK', 'LH_SHANK', 'RF_SHANK', 'RH_SHANK', 'LF_FOOT', 'LH_FOOT', 'RF_FOOT', 'RH_FOOT'
```

So in your case, the body indexing needs to be done w.r.t. articulation view as you read the Jacobian quantity from the articulation view. Can you please check this and see if the issue still persists?

**Comment by catachiii** [comment]
@Mayankm96 

Thanks for your clarification! Indeed, the index messed things up. 

I used the find_bodies function in rigid_object to find the index...
`    def find_bodies(self, name_keys: str | Sequence[str]) -> tuple[list[int], list[str]]:
        """Find bodies in the articulation based on the name keys.

        Please check the :meth:`orbit.utils.string_utils.resolve_matching_names` function for more
        information on the name matching.

        Args:
            name_keys: A regular expression or a list of regular expressions to match the body names.

        Returns:
            A tuple of lists containing the body indices and names.
        """
        return string_utils.resolve_matching_names(name_keys, self.body_names)`

This from my side will return the indices in rigid body view.

**Comment by Mayankm96** [comment]
Yes the data in Articulation follows rigid body view ordering. Only data from PhysX articulation view for links are in a different ordering. I have an MR under review right now which makes all this in the same indexing. It should get merged later today or this week.

**Comment by Mayankm96** [comment]
Merged the MR that should fix the issue: e3c40acf5e6433c1d88b54a6e9d45bd6d3e73152

Please reopen if the problem still persists.
